### ISSUE TITLE
Getting median fluorescence intensity (MFI) values and its quartile values from a specific channel

### ISSUE THREAD

**Issue by cjungR** [RGLab/flowWorkspace#212]
Hello,

I am trying to use flowWorkspace and QUALIFIER to get MFI values from flow data, but I can’t access the channels I want MFI values from. For example, using getQAStats() on a singlet population, I can only obtain MFI values for FSC-A and FSC-H (channels used to gated for singlets), but can't get values for the channel I want (like PE-A): 

```
> getQAStats(gs[[1]], isMFI = TRUE,  isRaw = TRUE, pops = c("/Cells/Single Cells"))

reading GatingHierarchy:Specimen_001_Tube_001.fcs_10000

   channel stain      stats        value                node          population
1:      NA    NA proportion 7.744804e-01 /Cells/Single Cells /Cells/Single Cells
2:      NA    NA      count 6.707000e+03 /Cells/Single Cells /Cells/Single Cells
3:   FSC-A    NA        MFI 8.268800e+04 /Cells/Single Cells /Cells/Single Cells
4:   FSC-H    NA        MFI 7.150200e+04 /Cells/Single Cells /Cells/Single Cells
```
In FlowJo I can find the MFI value for PE-A from this population without having actually gated for that particular channel. With flowWorkspace and QUALIFIER, it seems I need to manually gate for the channel in FlowJo to get MFI values for it. Do you know if it is possible to get MFI values from a channel without gating for it on FlowJo? 

Another question: is it possible to obtain quartile values for MFI using getQAStats()? Ie 25th quartile, 75th quartile values. 

Any help would be appreciated. Thank you for your time!

**Comment by gfinak** [comment]
@mikejiang 
I think this is something we should build into flowWorkspace. We could have a call that returns a variety of stats, we could even pass in a user defined function to compute stats for each population.

**Comment by gfinak** [comment]
Currently this is not possible without calling `getData()` on each population and summarizing the channel-level information, but I think it's worth adding this feature. 
Can you give the specific use case where channel information for channels other than those used for gating are of interest for QC for a specific population?

**Comment by mikejiang** [comment]
I have something called [getMFI](https://github.com/RGLab/cytoUtils/blob/master/R/getMFI.R#L18-L25) in `cytoUtils` . It is currently implemented at `R` level so could be a little slow.

**Comment by mikejiang** [comment]
closed by RGLab/cytoUtils#1
